### PR TITLE
Apply some Pylint fixes : Part 2

### PR DESCRIPTION
**Pull request by EmilyBourne**
Apply Pylint fixes and add docs to `pyccel.ast.builtins`.

### CODE REVIEW

File: pyccel/ast/builtins.py [hunk 1]
```diff
@@ -812,21 +950,27 @@
         if isinstance(x, (list, tuple)):
             x = PythonTuple(*x)
         elif not isinstance(x, (PythonTuple, PythonList)):
-            raise TypeError('Unknown type of  %s.' % type(x))
+            raise TypeError(f'Unknown type of {type(x)}.' )
         if not x.is_homogeneous:
-            types = ', '.join('{}({})'.format(xi.dtype,xi.precision) for xi in x)
+            types = ', '.join('{xi.dtype}({xi.precision})' for xi in x)
```
[the remark this answers is not in the document]
Comment (EmilyBourne) [reply pyccel/ast/builtins.py:955]: /bot accept code for errors

File: pyccel/ast/builtins.py [hunk 2]
```diff
@@ -840,25 +984,28 @@
         if isinstance(x, (list, tuple)):
             x = PythonTuple(*x)
         elif not isinstance(x, (PythonTuple, PythonList)):
-            raise TypeError('Unknown type of  %s.' % type(x))
+            raise TypeError(f'Unknown type of {type(x)}.' )
         if not x.is_homogeneous:
-            types = ', '.join('{}({})'.format(xi.dtype,xi.precision) for xi in x)
+            types = ', '.join(f'{xi.dtype}({xi.precision})' for xi in x)
```
[the remark this answers is not in the document]
Comment (EmilyBourne) [reply pyccel/ast/builtins.py:989]: /bot accept code for errors

File: pyccel/ast/numpyext.py [hunk 3]
```diff
@@ -181,10 +207,28 @@
...
+        """
+        Indicates whether the function can be applied elementwise.
+        
+        Indicates whether the function should be
+        called elementwise for an array argument
+        """
+        return True
+
 #=======================================================================================
 # TODO [YG, 13.03.2020]: handle case where base != 10
```
[the remark this answers is not in the document]
Comment (EmilyBourne) [reply pyccel/ast/numpyext.py]: /bot accept simple elemental definition

File: pyccel/ast/numpyext.py [hunk 4]
```diff
@@ -303,6 +368,16 @@
...
+    @property
+    def is_elemental(self):
+        """
+        Indicates whether the function can be applied elementwise.
+        
+        Indicates whether the function should be
+        called elementwise for an array argument
+        """
+        return True
+
```
[the remark this answers is not in the document]
Comment (EmilyBourne) [reply pyccel/ast/numpyext.py]: /bot accept simple elemental definition

[the remark this answers is not in the document]
Comment (EmilyBourne) [reply pyccel/ast/numpyext.py:380]: /bot accept simple elemental definition

**Review by yguclu — APPROVED** [review]

**Review by yguclu — APPROVED** [review]

File: pyccel/ast/numpyext.py [hunk 5]
```diff
@@ -181,10 +207,28 @@
...
 
+    @property
+    def is_elemental(self):
+        """
+        Indicates whether the function can be applied elementwise.
+        
+        Indicates whether the function should be
+        called elementwise for an array argument
+        """
+        return True
```
[the remark this answers is not in the document]
Comment (yguclu) [reply pyccel/ast/numpyext.py:218]: /bot accept simple elemental definition

File: pyccel/ast/numpyext.py [hunk 6]
```diff
@@ -303,6 +368,16 @@
...
```
[the remark this answers is not in the document]
Comment (yguclu) [reply pyccel/ast/numpyext.py:379]: /bot accept simple elemental definition

**Pull request merged**